packages/tldraw/src/lib/shapes/shared/useImageOrVideoAsset.ts

```diff
@@ -134,7 +136,15 @@ function resolveAssetUrl(
 		})
 }
 
-const resolveAssetUrlDebounced = debounce(resolveAssetUrl, 500)
+const debouncedFunctions = new Map<TLAssetId, typeof resolveAssetUrl & { cancel(): void }>()
+
+function getResolveAssetUrlDebounced(assetId: TLAssetId) {
```

Comment:
if we have two images pointing to the same asset (quite common - duplicating the image will do this, it was the original motivation for the asset resolution system) this will still cause issues as only one of the hooks will actually update. 

Instead of declaring this outside of the component, what if we added it in the component before the main `useLayoutEffect`:

```tsx
const resolveAssetUrlDebounced = useMemo(
    () => debounce(resolveAssetUrl, 500),
    [],
)
```

Reply:
much simpler! love it